### PR TITLE
Ywang2/update auto configuration api

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Support ov::hint::model_priority config key

### Tickets:
 - https://jira.devtools.intel.com/browse/CVS-76377

### CODE REVIEW

**Review by wangleis — APPROVED** [review]

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -61,6 +62,7 @@ namespace {
                     res.push_back(PluginConfigParams::KEY_PERF_COUNT);
                     res.push_back(PluginConfigParams::KEY_EXCLUSIVE_ASYNC_REQUESTS);
                     res.push_back(MultiDeviceConfigParams::KEY_AUTO_NETWORK_PRIORITY);
```
Comment (myshevts) [inline src/plugins/auto/plugin.cpp:64]: my understanding that the model_priority should replace (or alias?) the ov::hint:model_priority.
cc  @apankratovantonp

Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp:64]: Yes, will update soon.

File: src/tests/unit/auto/key_model_priority_test.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    ASSERT_NO_THROW(priority = plugin->GetConfig(MultiDeviceConfigParams::KEY_AUTO_NETWORK_PRIORITY, opt).as<std::string>());
+    EXPECT_EQ(priority, expectedPriority);
+}
+
+// ConfigParams details
+const std::vector<ConfigParams> testConfigs = {
+                                               ConfigParams {{{"MODEL_PRIORITY", "LOW"}}, "2"},
+                                               ConfigParams {{{"MODEL_PRIORITY", "MEDIUM"}}, "1"},
+                                               ConfigParams {{{"MODEL_PRIORITY", "HIGH"}}, "0"}
+                                              };
```
Comment (ilya-lavrenov) [inline src/tests/unit/auto/key_model_priority_test.cpp:77]: I suppose we need to write tests using new API?

File: src/plugins/auto/plugin.cpp [hunk 3]
```diff
@@ -182,6 +184,25 @@ void MultiDeviceInferencePlugin::SetConfig(const std::map<std::string, std::stri
...
+            _config[ov::hint::model_priority.name()] = "HIGH";
+            break;
+        case 1:
+            _config[ov::hint::model_priority.name()] = "MEDIUM";
+            break;
+        case 2:
+            _config[ov::hint::model_priority.name()] = "LOW";
+            break;
+        default:
+            throw ov::Exception{"Unsupported log level"};
```
Comment (ilya-lavrenov) [inline src/plugins/auto/plugin.cpp:203]: why log level?

**Review by myshevts — APPROVED** [review]
pls get approvale from @apankratovantonp as well

File: src/tests/unit/auto/key_model_priority_test.cpp (hunk 2 above)
Comment (yangwang201911) [reply src/tests/unit/auto/key_model_priority_test.cpp]: Updated.

File: src/plugins/auto/plugin.cpp (hunk 3 above)
Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp]: removed.

File: src/tests/unit/auto/key_model_priority_test.cpp (hunk 2 above)
Comment (wangleis) [reply src/tests/unit/auto/key_model_priority_test.cpp]: This PR will enable ov::hint:model_priority based on current implementation of KEY_AUTO_NETWORK_PRIORITY before Chinese New Year. Test case only cover mapping between ov::hint:model_priority and KEY_AUTO_NETWORK_PRIORITY.

After Chinese New Year, we will create another PR to replace KEY_AUTO_NETWORK_PRIORITY include test case using new API.

Comment (yangwang201911) [reply src/tests/unit/auto/key_model_priority_test.cpp]: @wangleis , the key KEY_AUTO_NETWORK_PRIORITY has already replaced in the latest commit and also added the corresponding test case for the key ov::hint:model_priority. Maybe we can enable the AUTO unit model priority test case on the  another PR?

**Comment by apankratovantonp** [comment]
Please also update plugin to use 
- `ov::device::priorities`, 
- `ov::log::level`, 
- `ov::enable_profiling`

**Comment by peterchen-intel** [comment]
> Please also update plugin to use
> 
> * `ov::device::priorities`,
> * `ov::log::level`,
> * `ov::enable_profiling`

@apankratovantonp Will create another PR for the above update

File: src/tests/functional/plugin/gpu/shared_tests_instances/behavior/plugin/configuration_tests.cpp [hunk 4]
```diff
@@ -96,13 +92,7 @@ namespace {
...
             {{InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES ,
                  CommonTestUtils::DEVICE_GPU + std::string(",") + CommonTestUtils::DEVICE_CPU},
-                    {InferenceEngine::PluginConfigParams::KEY_LOG_LEVEL, "NAN"}},
-            {{InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES ,
-                 CommonTestUtils::DEVICE_GPU + std::string(",") + CommonTestUtils::DEVICE_CPU},
-                {InferenceEngine::MultiDeviceConfigParams::KEY_AUTO_NETWORK_PRIORITY, "-1"}},
-            {{InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES ,
-                 CommonTestUtils::DEVICE_GPU + std::string(",") + CommonTestUtils::DEVICE_CPU},
-                {InferenceEngine::MultiDeviceConfigParams::KEY_AUTO_NETWORK_PRIORITY, "should be int"}}
+                    {InferenceEngine::PluginConfigParams::KEY_LOG_LEVEL, "NAN"}}
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/gpu/shared_tests_instances/behavior/plugin/configuration_tests.cpp:95]: why `NAN`? I suppose should be https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/openvino/runtime/properties.hpp#L316

**Pull request merged**